en/docs/install-and-setup/setup/api-controller/managing-microgateways/managing-microgateways-with-ctl.md

```diff
@@ -0,0 +1,293 @@
+# Managing Microgateways
+
+WSO2 API Controller, **apictl** can be used as a developer CLI tool for Microgateway. It enables interacting with one or more Microgateways without having to perform the following tasks via the WSO2 API Manager.
+
+- Deploy an API
+- View the list of deployed APIs
+- Undeploy an API
+
+!!! info
+    **Before you begin** 
+
+    -  Download the Microgateway distribution from the [github release page's](https://github.com/wso2/product-microgateway/releases) and follow the Quick Start Guide attached with the release.
+
+    -  Make sure the CTL is downloaded and initialized, if not follow the steps in [Download and Initialize the CTL Tool]({{base_path}}/install-and-setup/setup/api-controller/getting-started-with-wso2-api-controller/#download-and-initialize-the-ctl-tool). Following the guide upto [Check the version of the CTL]({{base_path}}/install-and-setup/setup/api-controller/getting-started-with-wso2-api-controller/#check-the-version-of-the-ctl) will be suffecient to continue. 
+
+
+## Add a Microgateway Environment
+
+A microgateway here refers to the entire cluster which includes all three components: adapter, router and enforcer. By adding a microgateway environment, the url to the adapter will be stored in a config file, and therefore enables login. 
+
+Once you have added a microgateway environment, and logged in, other commands such as `apictl mg deploy api` can be used by only specifying an environment. This allows you use such commands without entering the microgateway adapter url, username and password everytime.
+
+-   **Command**
+    ```bash
+    apictl mg add env <environment-name> --host <adapter-url> -k
+    ```
+
+    !!! info
+        **Flags:**
+        
+        -    Mandatory :     
+            `--host` or `-c` : The microgateway adapter host url with port  
+
+    !!! example
+        ```bash
+        apictl mg add env prod --host  https://localhost:9443 
+        ```
+
+## Log in to a Microgateway Environment
+
+Log in will get an access token from the microgateway adapter and store it for further commands.
+
+-   **Command**
+    ```go
+    apictl mg login <environment-name> -k
+    ```
+    ```go
+    apictl mg login <environment-name> -u <username> -k
+    ```
+    ```go
+    apictl mg login <environment-name> -u <username> -p <password> -k
+    ```
+    ```go
+    cat <file-with-the-password> | apictl mg login <environment-name> -u <username> --password-stdin -k
+    ``` 
```

Comment:
Shall we move this example inside the tip below under the point "If you include the flag `--password-stdin`, you can provide the password in a file as shown above", WDYT?
(When I was first reading it took some time to understand what this is. But when I look at the tip, the point have explained this. So, I would have save my time if directly look into the tip, but I didn't know that there is something related to this example in the upcoming tip.

Reply:
Thanks for this feedback. I'll move this into the tip.